app/assets/v2/js/pages/bounty_details.js

```diff
@@ -204,30 +206,103 @@ var pendingChangesWarning = function(issueURL, last_modified_time_remote, now){
-                var bounty = web3.eth.contract(bounty_abi).at(bounty_address());
-                setTimeout(function(){
-                    bounty.bountydetails.call(issueURL, function(error, result){
-                        if(error){
-                            setTimeout(check_for_bounty_changed_updates_web3, 1000);
+
+        // Only run this after the transaction is confirmed to be on the blockchain
+        var getBountyId = function (callback) {
+            // Get total number of bounties on the contract
+            if (localStorage['bountyId']) {
+                callback(null, localStorage['bountyId']);
+            }
+            var transactionInfo;
+            var bountiesLength;
+            var bountyId;
+            // var bounty = web3.eth.contract(bounty_abi).at(bounty_address());
+            var bounty = web3.eth.contract(bounty_abi).at(bounty_address());  //hardcode to test
+
+            bounty.getNumBounties(function(error, result){
+                console.log("Callback is now running...");
+                if (error){
+                    console.error(error);
+                    return;
+                }
+                bountiesLength = parseInt(result, 10);
+                console.log('Total bounties length: ' + bountiesLength.toString());
+                // Get bounty issuer
+                var i = bountiesLength - 1;
+                function getBountyLoop (i) {
+                    console.log('Trying bountyId: ' + i);
+                    bounty.getBounty(i, function(error, result2) {
+                        if (error) {
                             console.error(error);
+                            return;
+                        }
+                        issuer = result2[0];
+                        var bountyDetails = JSON.parse(localStorage['bountyDetails']);
+                        console.log('issuer from stdb: ' + issuer);
+                        console.log('account from bountyDetails:' + localStorage['bountyDetails'].split(",")[2])
+                        // compare issuer to the submitting address
+                        if (issuer == bountyDetails[2]) {
+                            bounty.getBountyData(i, function(error, result3) {
+                                if (error) {
+                                console.error(error);
+                                return;
+                                }
+                                dataHash = result3;
+                                console.log('dataHash from stdb: ' + dataHash);
+                                console.log('dataHash from cache: ' + localStorage['dataHash']);
+                                // compare resulting datahash from std bounties to cached dataHash
+                                if (dataHash == localStorage['dataHash']) {
+                                    var bountyId = i;
+                                    console.log('Found matching bountyId: '+ bountyId);
```

Comment:
well have to remove all the console logs before shipping

Reply:
Can do that.  Some were in there already but I did add a bunch to help with debugging and such.